### PR TITLE
Fix issue #3134: error in returning connectors from functions

### PR DESCRIPTION
**Pull request by pubudu91**
This PR fixes the issue #3134

### CODE REVIEW

File: modules/ballerina-core/src/test/java/org/ballerinalang/model/connectors/ConnectorTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        BValue[] args = {new BString("John"), new BInteger(20)};
+
+        BValue[] returns = BLangFunctions.invokeNew(programFile, "testConnectorReturn", args);
+
+        Assert.assertEquals(returns.length, 1);
+        Assert.assertSame(returns[0].getClass(), BString.class);
+
+        Assert.assertEquals(returns[0].stringValue(), "Username: BallerinaJohn, Age: 30");
+    }
+}
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/test/java/org/ballerinalang/model/connectors/ConnectorTest.java]: Shall we add a test case which returns connector in a ballerina action.

File: modules/ballerina-core/src/test/java/org/ballerinalang/model/connectors/ConnectorTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.codegen.ProgramFile;
+import org.ballerinalang.util.program.BLangFunctions;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * Test the usage of connectors in various scenarios
+ */
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/test/java/org/ballerinalang/model/connectors/ConnectorTest.java]: Add @ since annotation with the version

Comment (pubudu91) [inline modules/ballerina-core/src/test/java/org/ballerinalang/model/connectors/ConnectorTest.java]: Ack. will do.

File: modules/ballerina-core/src/test/resources/lang/connectors/connector-test.bal [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    Foo foo = create Foo(name, age);
+    return foo;
+}
+
+function testConnectorReturn (string name, int age) (string) {
+    Foo conn = getFooConnector(name, age);
+    string userName = conn.getUserName();
+    int userAge = conn.getAge();
+    return "Username: " + userName + ", Age: " + userAge;
+}
```
Comment (anupama-pathirage) [inline modules/ballerina-core/src/test/resources/lang/connectors/connector-test.bal]: We need to add test case which is taking an connector as argument to function as well. Refer the [discussion](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/ballerina-dev/uasy0zaPEZI/IturA6iMCQAJ)

**Comment by muthulee** [comment]
@pubudu91 - Shall we add a test case?

**Comment by pubudu91** [comment]
@muthulee The rewrite of the core may render this PR obsolete. If so, will send a new PR with juts the test cases.

**Comment by pubudu91** [comment]
Closing this since PR #3700 makes this PR obsolete

**Pull request closed without merging**